msgpack-core/src/main/java/org/msgpack/core/buffer/MessageBuffer.java

```diff
@@ -351,6 +252,9 @@ else if(directByteBufferClass.isInstance(buffer.base)) {
      */
     MessageBuffer(ByteBuffer bb) {
         if(bb.isDirect()) {
+            if(isUniversalBuffer) {
+                throw new IllegalStateException("Cannot create MessageBuffer from DirectBuffer");
```

Comment:
This code doesn't seem to be called since the constructor of MessageBufferU will be called if `isUniversalBuffer` is true.